src/services/contract-service.js

```diff
@@ -43,8 +44,9 @@ class ContractService {
       try {
         this[name].networks = Object.assign(
           {},
-          this[name].networks,
-          options.contractAddresses[name]
+          ((ContractAddresses[name]|{}).network|{}), // start with hard coded addresses
```

Comment:
Can you explain why you are using `|` here?

I see that it's the bitwise OR operator. But why is it being used? At a glance it seems like it might produce unintended behavior. For example, `undefined|{}` evaluates to `0`.

Even if it does behave as intended, I'm hesitant to use obscure operators. Maybe it's just me but I've never seen anyone do this in Javascript before.

Why not just use the regular OR (`||`)?

Reply:
Thanks! Fixed!